### PR TITLE
[WIP] mpd: 0.20.23 -> 0.21.3

### PR DESCRIPTION
**Pull request by pbogdan**
https://raw.githubusercontent.com/MusicPlayerDaemon/MPD/v0.21.3/NEWS

###### Motivation for this change

Update to the latest release.

(Among other changes) upstream switched the build system to meson, and as neither the flag names nor semantics of enabling / disabling flags match the autoconf build this makes the diff rather large.

Also since I had to go through meson build options anyway to map the current features to their new flag names I added a bunch of new flags. Probably doesn't make sense to keep all of them but figured might show what's available in the new build system. Feature-wise this build should match the current derivation and shouldn't enable / disable any new features.

I also may have inadvertently broken support for platforms other that x86_64-linux as I'm unable to test outside that platform.

Finally, this isn't tested heavily. My own personal setup is extremely simple and I'm not even sure how to check many of the features (that are enabled by default).

Anyway, had these changes sitting around for a while so figured might open a PR in case it's helpful at all and helps with the update. Should at least take care of the tediousness of mapping things to the new build system.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by pbogdan** [comment]
@GrahamcOfBorg build mpd

File: nixos/modules/services/audio/mpd.nix [hunk 1]
```diff
@@ -144,6 +146,7 @@ in {
   ###### implementation
 
   config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.mpd ];
```
Comment (Mic92) [inline nixos/modules/services/audio/mpd.nix]: Why is this required?

**Comment by Mic92** [comment]
@GrahamcOfBorg build mpd

File: nixos/modules/services/audio/mpd.nix (hunk 1 above)
Comment (pbogdan) [reply nixos/modules/services/audio/mpd.nix]: Not _strictly_ required but it installs the man pages and the documentation which I thought might be useful.

**Review by infinisil — COMMENTED** [review]
Tbh I'm not a big fan of having a hundred options for configuring packages. Why not just have a minimal and a full version? Nix doesn't need to reproduce every possible option in the build system. I know other (popular?) nixpkgs packages do this too and I wonder why.

**Comment by Mic92** [comment]
Yeah it should be possible to use `overrideAttrs` to set mesonFlags and additional dependencies yourself.

**Comment by pbogdan** [comment]
@Infinisil I'm not too fond of that either but stuck with the existing style / convention of the existing expression. Personally I wouldn't be even opposed to removing them all and just providing `mpd` with the feature set matching current default configuration though I suppose some users may rely on some flags being present in the current expression?

@Mic92 

>  Also how did you test all these combination of flags do even work?

I have not tested every possible permutation but for each flag I checked that at least the configure phase finds the required dependencies and proceeds to build. Enabling all of the features does result in a build failure (`chromaprint` support and one other feature that I can't remember right now would need fixing) but I didn't look into it too hard as I wasn't sure we will want to actually keep it.
The default configuration does work for me but as I said my personal use case is very simple.
